### ISSUE TITLE
InheritedPermissions caused performance regression on large File dataset

### ISSUE THREAD

**Issue by maxime-rainville** [silverstripe/silverstripe-assets#447]
This [snippet of code](https://github.com/silverstripe/silverstripe-asset-admin/blob/48d0304799f5f203ec621f04bcd41db6cf82f4cd/code/GraphQL/FolderTypeCreator.php#L212-L216) causes a can view check on all the files in a folder and the fact that it has been moved around in this commit: [d3968ad](https://github.com/silverstripe/silverstripe-asset-admin/commit/d3968adcbdb759cb20571865af3b6356c8922cde)

If you have thousands of files in a folder this will potentially caused your request to crash because of this [query in `InheritedPermissions.php`](https://github.com/silverstripe/silverstripe-framework/blob/51a882eca8292f3814d5853e58b840d4417d8073/src/Security/InheritedPermissions.php#L298-L299)
```php
$stageRecords = Versioned::get_by_stage($this->getBaseClass(), $stage)
    ->byIDs($ids);
```

### Versions affected

silverstripe/asset-admin 1.7.1
silverstripe/framework 4.7.3

### Replication details
"We have almost 80k of assets in one folder and the code wants to check permission for each one of them even though the pagination is set to 50"

### Implementation notes
- Looks like we will be adding a limit on the check - need to work out when
- Noting that we expect the number of results and pagination to not to match the actual results - this is a known issue

### PRs
- [x] https://github.com/silverstripe/silverstripe-asset-admin/pull/1220

**Comment by maxime-rainville** [comment]
@brynwhyman Won't affect many clients, but will potentially make asset-admin unusable for those with lots of files.

**Comment by mfendeksilverstripe** [comment]
Some more details:

There two issues here:

* The lookup by IDs will fail with large number of assets as the query placeholder limit may be reached
* Loop [in the permission check](https://github.com/silverstripe/silverstripe-framework/blob/51a882eca8292f3814d5853e58b840d4417d8073/src/Security/InheritedPermissions.php#L405) will exhaust memory when trying to load all asset models into memory

Some suggestions:

* do we really need to check permissions of all assets to display a single page of assets within a folder?
* we could potentially use [chunkedFetch](https://docs.silverstripe.org/en/4/developer_guides/model/lists/#chunkedFetch) feature which is available in `4.8`

**Comment by brynwhyman** [comment]
https://github.com/silverstripe/silverstripe-asset-admin/issues/1207 reported a similar issue and was closed in lieu of this issue. The report there noted file sizes of >9000 files which is significantly lower than the originally reported "80,000 files".

**Comment by emteknetnz** [comment]
PR https://github.com/silverstripe/silverstripe-asset-admin/pull/1220

**Comment by maxime-rainville** [comment]
@mfendeksilverstripe Just released 1.8.2 if you want to swap out your fork https://github.com/silverstripe/silverstripe-asset-admin/releases/tag/1.8.2
